src/parser/tera.pest

```diff
@@ -78,7 +79,7 @@ dotted_square_bracket_ident = @{
 
 // boolean first so they are not caught as identifiers
 basic_val  = _{ boolean | test | macro_call | fn_call | dotted_square_bracket_ident | float | int | string }
-basic_op   = _{ op_plus | op_minus | op_times | op_slash | op_modulo }
+basic_op   = _{ op_plus | op_minus | op_times | op_slash | op_modulo | op_tilde }
```

Comment:
This allows `~` in operations which is not great, ie `1 ~ 2` would be considered a concat but doesn't mean anything in the real world.

I view string concat more like `str_concat = { (string | dotted_square_bracket_ident) ~ "~" ~  string | dotted_square_bracket_ident)}` (maybe with `+` somewhere if we want multiple concat) and that `str_concat` rule being part of `basic_val`.

I haven't thought deeply about it so there might be some unforeseen issues with that though

Reply:
I'm struggling with pest and I think I've found all of the dead ends with the suggested approach. If possible, I'd like to have a synchronous discussion on this.

Of note: the Jinja parser in Python appears to combine concatenation with the math operation parsing. https://github.com/pallets/jinja/blob/master/jinja2/parser.py#L494-L524

Reply:
let me do a spike on my end first since i didn't really think things through in terms of code yet

Reply:
It was easier than I thought, #293